upgrade/runner.py

```diff
@@ -245,3 +250,61 @@ def product_upgrade(product):
             execute(foreman_debug, 'satellite_{}'.format(sat_host),
                     host=sat_host)
             raise
+
+
+# Fabric task
+def validate_and_create_product_templates():
+    """Task to do a sanity check on the satellite and capsule and then
+    create their templates after z-stream upgrade
+
+    Environment variables required to run upgrade on RHEVM Setup and will be
+    fetched from Jenkins:
+    ----------------------------------------------------------------------
+
+    RHEV_SAT_HOST
+        The rhevm satellite hostname to run upgrade on
+    RHEV_CAP_HOST
+        The rhevm capsule hostname to run upgrade on
+    RHEV_STORAGE
+        The storage domain on the rhevm used to create templates
+    RHEV_SAT_IMAGE
+        The satellite Image from which satellite instance will be created
+    RHEV_CAP_IMAGE
+        The capsule Image from which capsule instance will be created
+    RHEV_SAT_INSTANCE
+        The satellite instance name in rhevm of which template is to be
+        created, generally the upgraded box
+    RHEV_CAP_INSTANCE
+        The capsule instance name in rhevm of which template is to be
+        created, generally the upgraded box
+    """
+    # Get the instances name, specified in the jenkins job
+    sat_instance = os.environ.get('RHEV_SAT_INSTANCE')
+    cap_instance = os.environ.get('RHEV_CAP_INSTANCE')
+    cluster = 'Default'
+    storage = os.environ.get('RHEV_STORAGE')
+    if sat_instance and cap_instance:
+        sat_host = os.environ.get('RHEV_SAT_HOST')
+        new_sat_template = os.environ.get('RHEV_SAT_IMAGE') + "_new"
+        cap_host = os.environ.get('RHEV_CAP_HOST')
+        new_cap_template = os.environ.get('RHEV_CAP_IMAGE') + "_new"
+        if check_necessary_env_variables_for_upgrade('capsule'):
+            execute(check_ntpd, host=sat_host)
+            execute(katello_restart, host=sat_host)
+            execute(check_capsule, cap_host, host=sat_host)
+            execute(check_ntpd, cap_host)
+            execute(katello_restart, cap_host)
+            try:
+                thread.start_new_thread(create_rhevm_template,
+                                        (sat_instance,
```

Comment:
Code:
> def print_name(threadname,delay):
     count=0
     while count<5:
          time.sleep(delay)
          count = count + 1
          print threadname

Output: 

> thread.start_new_thread(print_name,('mythread',1,))
139623866775296
 mythread
mythread
mythread
mythread
mythread

This is a basic way of implementing thread , it could be done better. @rochacbruno @renzon any thoughts?